### PR TITLE
fix(artifacts): fixes multiple http base providers

### PR DESCRIPTION
**Pull request by ethanfrogers**
when enabling multiple http based artifact providers (http + gitlab +
github) clouddriver filed to start stating that multiple beans were
found.

### CODE REVIEW

**Comment by ethanfrogers** [comment]
here's the error message that i saw:
```
***************************
APPLICATION FAILED TO START
***************************

Description:

Field gitHubOkHttpClient in com.netflix.spinnaker.clouddriver.artifacts.github.GitHubArtifactConfiguration required a single bean, but 2 were found:
    - httpOkHttpClient: defined by method 'httpOkHttpClient' in class path resource [com/netflix/spinnaker/clouddriver/artifacts/http/HttpArtifactConfiguration.class]
    - gitlabOkHttpClient: defined by method 'gitlabOkHttpClient' in class path resource [com/netflix/spinnaker/clouddriver/artifacts/gitlab/GitlabArtifactConfiguration.class]
```

**Comment by ethanfrogers** [comment]
@ttomsu @lwander PTAL.

**Comment by ethanfrogers** [comment]
we'll want to pull this into 1.8.x as well.

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/bitbucket/BitbucketArtifactConfiguration.java [hunk 1]
```diff
@@ -53,7 +53,6 @@ BitbucketArtifactProviderProperties bitbucketArtifactProviderProperties() {
   @Autowired
   ArtifactCredentialsRepository artifactCredentialsRepository;
 
-  @Autowired
   OkHttpClient bitbucketOkHttpClient;
```
Comment (ttomsu) [inline clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/bitbucket/BitbucketArtifactConfiguration.java]: I don't think these fields are needed anymore, right? Across all of the ArtifactConfiguration classes.

Comment (ethanfrogers) [reply clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/bitbucket/BitbucketArtifactConfiguration.java]: ah, i see! i'll pull them out.

**Pull request merged**